pkg/whitesource/whitesource.go

```diff
@@ -102,21 +105,16 @@ func (s *System) GetProductsMetaInfo() ([]Product, error) {
 		RequestType: "getOrganizationProductVitals",
 	}
 
-	respBody, err := s.sendRequest(req)
+	err := s.sendRequestAndDecodeJSON(req, &wsResponse)
 	if err != nil {
 		return wsResponse.ProductVitals, errors.Wrap(err, "WhiteSource request failed")
 	}
 
-	err = json.Unmarshal(respBody, &wsResponse)
-	if err != nil {
-		return wsResponse.ProductVitals, errors.Wrap(err, "failed to parse WhiteSource response")
-	}
-
 	return wsResponse.ProductVitals, nil
 }
 
-// GetMetaInfoForProduct retrieves meta information for a specific WhiteSource product
-func (s *System) GetMetaInfoForProduct(productName string) (Product, error) {
+// GetProductByName retrieves meta information for a specific WhiteSource product
+func (s *System) GetProductByName(productName string) (Product, error) {
```

Comment:
incompatible change

Reply:
No, there were two functions, GetMetaInfoForProduct and GetProductByName, which did the exact same thing. I removed one instance, and kept GetProductByName.

Reply:
shouldn't we stick with GetMetaInfoForProduct since this is closer to the api used - but nevermind ...

Reply:
Hm, you *get a `Product`* object. And there is the symmetry with `GetProductByToken()`.